dlp/snippets/deid/deid_table_fpe.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package deid
+
+// [START dlp_deidentify_table_fpe]
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+)
+
+// deidentifyTableFPE de-identifies table data with format preserving encryption.
+func deidentifyTableFPE(w io.Writer, projectID string, kmsKeyName, wrappedAesKey string) error {
+	// projectId := "your-project-id"
+	/* kmsKeyName := /* keyFileName :=  "projects/YOUR_PROJECT/"
+	   + "locations/YOUR_KEYRING_REGION/"
+	   + "keyRings/YOUR_KEYRING_NAME/"
+	   + "cryptoKeys/YOUR_KEY_NAME"
+	*/
+	// wrappedAesKey := "YOUR_ENCRYPTED_AES_256_KEY"
+	// table := "your-table"
+
+	// define your table.
+	row1 := &dlppb.Table_Row{
+		Values: []*dlppb.Value{
+			{Type: &dlppb.Value_StringValue{StringValue: "11111"}},
+			{Type: &dlppb.Value_StringValue{StringValue: "2015"}},
+			{Type: &dlppb.Value_StringValue{StringValue: "$10"}},
+		},
+	}
+
+	row2 := &dlppb.Table_Row{
+		Values: []*dlppb.Value{
+			{Type: &dlppb.Value_StringValue{StringValue: "22222"}},
+			{Type: &dlppb.Value_StringValue{StringValue: "2016"}},
+			{Type: &dlppb.Value_StringValue{StringValue: "$20"}},
+		},
+	}
+
+	row3 := &dlppb.Table_Row{
+		Values: []*dlppb.Value{
+			{Type: &dlppb.Value_StringValue{StringValue: "33333"}},
+			{Type: &dlppb.Value_StringValue{StringValue: "2016"}},
+			{Type: &dlppb.Value_StringValue{StringValue: "$15"}},
+		},
+	}
+
+	table := &dlppb.Table{
+		Headers: []*dlppb.FieldId{
+			{Name: "Employee ID"},
+			{Name: "Date"},
+			{Name: "Compensation"},
+		},
+		Rows: []*dlppb.Table_Row{
+			{Values: row1.Values},
+			{Values: row2.Values},
+			{Values: row3.Values},
+		},
+	}
+
+	// client operations.
+	ctx := context.Background()
+
+	// Initialize a client once and reuse it to send multiple requests. Clients
+	// are safe to use across goroutines. When the client is no longer needed,
+	// call the Close method to cleanup its resources.
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("dlp.NewClient: %v", err)
```

Comment:
Use %w for error wrapping.

Reply:
- again based on this [comment](https://github.com/GoogleCloudPlatform/golang-samples/pull/3166#discussion_r1238023687) I have made the changes, still let me know the final decision. Thank you.

Reply:
Ah, I think there's a misunderstanding. The intention of my previous comments was to **just return the error**. There's no need to insert it into a formatting string.

Reply:
Discussed this offline with @telpirion. Returning the error directly is fine, but if you do use `fmt.Errorf` you should use `%w` rather than `%v` as your formatting verb so that the caller can access the wrapped error.

Reply:
So just returning the `err` here to make consistency across the samples.